### PR TITLE
Feature/tech debt reduce complexity

### PR DESCRIPTION
**Pull request by kpeacocke**
# Pull Request Template

## 🌿 Gitflow Branch Check

**Source branch**: `feature/tech-debt-reduce-complexity`
**Target branch**: `develop`

Please verify your branch follows gitflow conventions:
- ✅ `feature/*` → targets `develop`
- ✅ `bugfix/*` → targets `develop`
- ✅ `release/*` → targets `main`
- ✅ `hotfix/*` → targets `main`

_The gitflow workflow will automatically validate this._

---

## 📋 Summary

Brief description of what this PR does and why.

**Type of Change** (check all that apply):
- [x] 🐛 Bug fix (non-breaking change that fixes an issue)
- [ ] ✨ New feature (non-breaking change that adds functionality)
- [x] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📚 Documentation update
- [x] 🧪 Test improvements
- [x] 🔧 Refactoring (no functional changes)
- [x] 📦 Dependencies update
- [x] 🚀 Performance improvement

## 🎯 Related Issues

Fixes: #20 #19 #18 #17 #16 #15 #14 #13 #12 #11 #10 #9 #8 #7 #6 

### Core Changes
- Migrate from uv to poetry
- refactor to remove completely 

### Files Modified
- `souschef/server.py` - [description of changes]
- `souschef/cli.py` - [description of changes]
- `tests/test_*.py` - [description of test changes]
- `README.md` - [description of documentation changes]

### Test Coverage
- [x] Unit tests added/updated for new functionality
- [x] Integration tests added/updated
- [x] Property-based tests added if applicable
- [x] All existing tests pass
- [x] Code coverage maintained or improved

### Manual Testing

```bash
# Commands run for testing
souschef-cli recipe examples/recipe.rb
```

**Test Results**:
```
Executing task: poetry run pytest 

......................................................................................................................... [ 34%]
......................................................................................................................... [ 68%]
.................................................................................................................         [100%]
======================================================== tests coverage =========================================================
________________________________________ coverage: platform linux, python 3.14.2-final-0 ________________________________________

Name                   Stmts   Miss  Cover   Missing
----------------------------------------------------
souschef/__init__.py       0      0   100%
souschef/cli.py          182     21    88%   140-141, 200-205, 274-275, 293-294, 314-315, 332-333, 349-350, 383, 390-391, 395
souschef/server.py      2612    420    84%   40-41, 62-63, 174, 612, 632, 1163, 1282, 1293-1294, 1319-1320, 1386-1387, 1459, 1524, 1654, 1725-1730, 1885-1886, 1911-1913, 1918-1920, 1972-1974, 2126, 2246, 2326, 2460, 2508, 2559, 2588, 2605-2608, 2703, 2798, 2859-2861, 2865-2867, 2877-2879, 2900, 2921, 3001-3033, 3049-3080, 3105-3108, 3721-3722, 3746-3747, 3771, 3784-3785, 3844-3845, 3877-3878, 3928-3929, 3995-3996, 4026, 4058-4059, 4068-4069, 4097, 4122-4123, 4166-4167, 4213-4214, 4223-4224, 4254-4256, 4282-4283, 4311, 4316, 4321, 4346-4348, 4353-4363, 4377-4379, 4399, 4404, 4410, 4451, 4469-4479, 4513, 4518, 4539-4540, 4588-4616, 4622, 4656-4680, 4710, 4715, 4727, 4737-4760, 4813, 4820-4823, 4856, 4889-4890, 4940-4980, 5004, 5012, 5019-5028, 5048, 5053, 5061, 5071-5092, 5153-5154, 5206-5207, 5275-5276, 5337-5338, 5369-5383, 5395-5396, 5401, 5407, 5441, 5640-5648, 5670-5686, 5698, 5705-5727, 5742, 5775, 5814, 5885-5886, 6042-6043, 6107-6108, 6156-6157, 6185-6191, 6234-6240, 6290-6303, 6644, 6695, 6803-6804, 6840-6856, 6905-6912, 6942-6952, 6962-6969, 6981, 6985, 6987, 7011, 7013, 7116-7117, 7184-7185, 7242-7243, 7312-7313, 7341-7355, 7389, 7393, 7397, 7404-7405, 7428, 7430, 7470, 7478, 7505-7511, 7519-7520, 7529-7532, 7568-7573, 7594, 7600, 7606, 7613, 7620, 7633-7638, 7674-7680, 7716, 7749, 7782-7785, 7944-7946, 7954, 7969-7975, 7983, 7989, 8009, 8015, 8020, 8063, 8081, 8110, 8129
----------------------------------------------------
TOTAL                   2794    441    84%

-------------------------------------------------------------------------------------------------- benchmark: 7 tests -------------------------------------------------------------------------------------------------
Name (time in us)                               Min                   Max                Mean             StdDev              Median                IQR            Outliers  OPS (Kops/s)            Rounds  Iterations
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
test_benchmark_conversion                    1.2910 (1.0)         57.4160 (1.0)        1.4516 (1.0)       0.5637 (1.0)        1.4170 (1.0)       0.0829 (1.0)      497;4535      688.9099 (1.0)      166668           1
test_benchmark_inspec_conversion           120.2090 (93.11)      719.1250 (12.52)    180.9623 (124.67)   26.8311 (47.59)    174.1670 (122.91)   17.8228 (214.87)    226;170        5.5260 (0.01)       3575           1
test_benchmark_custom_resource_parsing     124.7920 (96.66)    1,472.8330 (25.65)    190.5850 (131.30)   33.7972 (59.95)    185.8330 (131.14)   20.4162 (246.14)    316;298        5.2470 (0.01)       4905           1
test_benchmark_parse_attributes            141.5409 (109.64)     575.4170 (10.02)    175.3643 (120.81)   24.8431 (44.07)    168.7500 (119.09)   14.7812 (178.20)      53;52        5.7024 (0.01)        761           1
test_benchmark_template_parsing            142.2500 (110.19)     613.5830 (10.69)    206.4219 (142.21)   21.9530 (38.94)    200.9160 (141.79)   17.5835 (211.99)    387;224        4.8444 (0.01)       4300           1
test_benchmark_parse_recipe                203.8330 (157.89)     467.5830 (8.14)     225.9717 (155.67)   19.1962 (34.05)    220.5420 (155.64)   12.7500 (153.72)      80;61        4.4253 (0.01)       1078           1
test_benchmark_inspec_profile_parsing      587.9170 (455.40)   2,295.5000 (39.98)    799.3950 (550.71)   92.1814 (163.52)   780.0620 (550.50)   57.1670 (689.21)     100;91        1.2509 (0.00)        900           1
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

Legend:
  Outliers: 1 Standard Deviation from Mean; 1.5 IQR (InterQuartile Range) from 1st Quartile and 3rd Quartile.
  OPS: Operations Per Second, computed as 1 / Mean
355 passed in 7.16s
```

### Test Cases Covered
- [x] Normal/happy path scenarios
- [x] Edge cases and error conditions
- [x] Cross-platform compatibility
- [x] Performance with large inputs
- [x] Integration with existing features

## 📊 Performance Impact

If applicable, describe performance implications:

- **Memory usage**: No significant change / Reduced / Increased by X%
- **Processing time**: Benchmarks show X% improvement/regression
- **Scalability**: Handles larger cookbooks / No impact

## 🔒 Security Considerations

For changes that might have security implications:

- [x] Input validation added/improved
- [x] No secrets or credentials exposed
- [x] File path validation in place
- [x] Error messages don't leak sensitive information
- [x] Dependencies updated to secure versions

## 📖 Documentation

- [x] README.md updated if needed
- [ ] CLI help text updated
- [ ] Code comments added for complex logic
- [ ] Docstrings added/updated following Google style
- [ ] Examples added or updated

### Checklist (verify before submitting)
- [x] ✅ Code passes linting
- [x] ✅ Code is properly formatted
- [x] ✅ All tests pass
- [x] ✅ Type hints are complete and accurate
- [x] ✅ Docstrings follow Google style format
- [x] ✅ No warnings introduced (zero warnings policy)
- [x] ✅ Code follows existing patterns and style
- [x] ✅ Error handling is comprehensive

### Code Review Areas
Please pay special attention to:
- [ ] Resource-to-task conversion logic
- [ ] ERB-to-Jinja2 template conversion
- [ ] File path handling and security
- [ ] Error messages and user experience
- [x] Test coverage and edge cases

## ⚠️ Breaking Changes

If this PR introduces breaking changes, describe:

1. **What breaks**: uv gone, poetry for python management
2. **Migration path**: update all tirggers, scripts, etc. to poetry
3. **Deprecation period**: NOW, while we are in pre-release

## 🚀 Deployment Notes

Any special considerations for deployment:

- [ ] Database migrations required
- [ ] Configuration changes needed
- [x] Environment variable updates
- [x] Dependencies updated
- [ ] Version bump required

## 🔄 Reviewer Guidance

**Focus Areas for Review**:
- Code correctness and edge case handling
- Test coverage and quality
- Documentation accuracy
- Performance implications
- Security considerations

**Questions for Reviewers**:
- Is the Chef-to-Ansible mapping accurate?
- Are there edge cases I missed?
- Is the error handling appropriate?
- Should this be split into smaller PRs?

---

## 📋 Pre-Submit Checklist

Before submitting this PR, I have verified:

- [x] 🧪 All tests pass locally (`uv run pytest`)
- [x] 🎨 Code is linted and formatted (`uv run ruff check . && uv run ruff format .`)
- [x] 📊 Test coverage is maintained or improved
- [x] 📖 Documentation is updated where needed
- [x] 🔍 I have self-reviewed my code changes
- [x] 💭 Complex logic has explanatory comments
- [x] 🚨 No debug code or console.log statements left
- [x] 🔧 Commit messages follow conventional format

**By submitting this PR, I confirm that**:
- [x] My contribution is made under the project's MIT license
- [x] I have followed the project's coding standards and guidelines
- [x] I have read and agree to the [Code of Conduct](../CODE_OF_CONDUCT.md)

### CODE REVIEW

File: souschef/server.py [hunk 1]
```diff
@@ -1133,6 +1218,23 @@ def _convert_chef_resource_to_ansible(
     return task
 
 
+def _format_yaml_value(value: Any) -> str:
+    """Format a value for YAML output."""
+    import json
+
```
[the remark this answers is not in the document]
Comment (kpeacocke) [reply souschef/server.py]: Agreed. Updating to remove the redundant imports

File: souschef/server.py [hunk 2]
```diff
@@ -2057,30 +2353,115 @@ def _extract_recipe_variables(raw_content: str) -> dict[str, str]:  # noqa: C901
...
+    """Extract owner and group specifications from recipe content.
+
+    Args:
+        raw_content: Raw Chef recipe file content.
+
+    Returns:
+        Dictionary with file_owner and/or file_group keys if found.
+
+    """
+    import re
```
[the remark this answers is not in the document]
Comment (kpeacocke) [reply souschef/server.py]: Removing and updating PR

**Comment by kpeacocke** [comment]
> **Critical Issue**: The complete removal of `uv.lock` without regeneration will break dependency reproducibility and is a blocker for this PR.

No, we moved to poetry.
